Source/DafnyCore/DafnyOptions.cs

```diff
@@ -580,6 +580,16 @@ public enum IncludesModes {
           WarnShadowing = true;
           return true;
 
+        case "warnContradictoryAssumptions":
```

Comment:
Could you implement this `DafnyOptions.RegisterLegacyUi` using this method instead? My desire is for code in general to be grouped by feature. Having code for many different features inside `DafnyOptions` breaks this structure. Ideally, I would put the options `WarnContradictoryAssumptions` and `WarnRedundantAssumptions` in the `ProofDependencyWarnings` class, including the call to `DafnyOptions.RegisterLegacyUi`, and then more of the code for that feature is grouped by that file.

Reply:
Oh, definitely. That seems much better. I hadn't realized I could do that. :)